Client/Ecosia/Experiments/UnleashBingExperiment.swift

```diff
@@ -0,0 +1,30 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Core
+
+struct BingSearchExperiment {
+    
+    private init() {}
+    
+    static func incrementCounter() {
+        BingSearchExperimentCounter.increment()
+    }
+    
+    static func isEnabled() -> Bool {
+        Unleash.isEnabled(.bingSearch)
+    }
+    
+    static func makeBingSearchURLFromURL(_ sourceURL: URL) -> URL? {
+        let baseDomain = sourceURL.normalizedHost ?? URL.domain
+        let bingDomain = "bing.com"
+        let bingFullURLString = sourceURL.absoluteString.replacingOccurrences(of: baseDomain, with: bingDomain)
```

Comment:
I guess in this case the main objective is to change the url host, right?
How about something like:
```
let components = URLComponents(url: sourceURL.absoluteString, resolvingAgainstBaseURL: false)
urlComponents?.host = "bing.com"
return urlComponents?.url
```
It seems a bit safer to me and enables us to add extra query items to the url if necessary 🙂

Reply:
Maybe this function could even have some tests to illustrate the expected behaviour 🤔 just thinking out loud, nothing essential

Reply:
That has been indeed a good point. I was actually looking for something like that as a solution but then went after the "bulletproof one". Tested an made some amends 👍. Will do the following

```
        var urlComponents = URLComponents(url: sourceURL.absoluteURL, resolvingAgainstBaseURL: false)
        urlComponents?.host = "bing.com"
        return urlComponents?.url
```